x/tokenfactory/keeper/createdenom.go

```diff
@@ -64,3 +70,31 @@ func (k Keeper) validateCreateDenom(ctx sdk.Context, creatorAddr string, subdeno
 
 	return denom, nil
 }
+
+func (k Keeper) chargeForCreateDenom(ctx sdk.Context, creatorAddr string) (err error) {
+	// Send creation fee to community pool
+	creationFee := k.GetParams(ctx).DenomCreationFee
+	accAddr, err := sdk.AccAddressFromBech32(creatorAddr)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrUnableToCharge, "wrong creator address: %v", err)
+	}
+
+	params := k.GetParams(ctx)
+
+	if len(creationFee) > 0 {
+		feeCollectorAddr, err := sdk.AccAddressFromBech32(params.FeeCollectorAddress)
+		if err != nil {
+			return sdkerrors.Wrapf(types.ErrUnableToCharge, "wrong fee collector address: %v", err)
+		}
+
+		err = k.bankKeeper.SendCoins(
+			ctx,
+			accAddr, feeCollectorAddr,
+			creationFee,
+		)
+
+		return sdkerrors.Wrap(err, "unable to send coins to fee collector")
```

Comment:
I believe you missed the `if err != nil` check here. it works because of the following hidden behaviour:

```go
// If err is nil, this returns nil, avoiding the need for an if statement when
// wrapping a error returned at the end of a function
func Wrap(err error, description string) error {
```

I don't think we should leave it this way cause it's so much obscure and against the most common go practice ever: checking `if err != nil` before err return. I understand that this is a bit more concise, but imo it rudely disrupts the main golang dogma. wyt?

same above at the `createDenomAfterValidation` call